### PR TITLE
DOC - 7867 - Named Scope and Collections first examples

### PR DESCRIPTION
**Pull request by maria-robobug**
[DOC-7867](https://issues.couchbase.com/browse/DOC-7867), [GOCBC-1036](GOCBC-1036 )

This PR covers:

- Updating all sample code to use named scopes and collections instead of defaults (where possible).
- Removing any mentions of Developer Preview
- Updating Antora references of server 6.5/6.6 to 7.0
- Extracting concept-docs examples from adoc files and added tests
- Minor test updates

Please note because of how big the change is, I've split the commits by folder so that you can look at each change separately.

The final commit is simply updating the github workflow files with the 2.3 branch so automation can run.

**Note:** The code sample tests will fail as we are using named scopes and collections for most of the examples, which we expect since the SDK is not compatible with the official server 7.0-beta image. I have tested them locally and they do pass, so once merged they will also run against the latest internal server build and should go green then.

### CODE REVIEW

File: modules/concept-docs/pages/n1ql-query.adoc [hunk 1]
```diff
@@ -7,42 +7,33 @@
...
-	rows, err := cluster.Query(query, &gocb.QueryOptions{
-		Adhoc: false,
-	})
-	if err != nil {
-		panic(err)
-	}
-
-	for rows.Next() {
-	}
+QueryResult result = cluster.query(
```
Comment (chvck) [inline modules/concept-docs/pages/n1ql-query.adoc]: Is this meant to be here?

Comment (maria-robobug) [reply modules/concept-docs/pages/n1ql-query.adoc]: Nope! Silly mistake, I must have pasted that in by accident. Good spot, will remove this.

**Review by chvck — APPROVED** [review]
Looks good to me!

**Review by RichardSmedley — APPROVED** [review]
:+1:

File: modules/concept-docs/pages/n1ql-query.adoc [hunk 2]
```diff
@@ -7,42 +7,33 @@
...
-include::6.5@sdk:shared:partial$n1ql-queries.adoc[tag=intro]
+include::7.0@sdk:shared:partial$n1ql-queries.adoc[tag=intro]
 
 
 // Prepared Statements for Query Optimization
-include::6.5@sdk:shared:partial$n1ql-queries.adoc[tag=prepared]
+include::7.0@sdk:shared:partial$n1ql-queries.adoc[tag=prepared]
 
-NOTE: In Go SDK 2.1 and earlier, the `adhoc` parameter is `Adhoc`. 
+NOTE: In Go SDK 2.3 and earlier, the `adhoc` parameter is `Adhoc`.
```
Comment (osfameron) [inline modules/concept-docs/pages/n1ql-query.adoc:16]: Doesn't this mean that the parameter is *always* Adhoc?

Comment (osfameron) [reply modules/concept-docs/pages/n1ql-query.adoc:16]: ah no, now appears to be `adhoc` as below?

**Pull request merged**